widgets/data_editor.py

```diff
@@ -299,6 +299,14 @@ def item_selected(item):
 
         return combobox
 
+    def add_button_input(self, labeltext, text, function):
+        button = QtWidgets.QPushButton(self)
+        button.setText(text)
+        button.clicked.connect(function)
```

Comment:
For consistency with `add_dropdown_input()`, the following could be added to make use of all the space in the row:

```python
policy = button.sizePolicy()
policy.setHorizontalPolicy(QtWidgets.QSizePolicy.Expanding)
button.setSizePolicy(policy)
```